File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.c
char* addSpaces(char* s, int* spaces, int spacesSize) {
    int index = 0;
    int strIndex = 0;
    int strLen = strlen(s);
    int totalLen = spacesSize + strLen;
    char *string = (char*)malloc((totalLen+1) * sizeof(char));
    

    for(int i = 0; i < totalLen; i++){
        
        if(index < spacesSize && strIndex == spaces[index]){
            
            string[i++] = ' ';
            index++;
            
        }
        
        string[i] = s[strIndex++]; 
    }
    string[totalLen] = '\0';
    return string;
}

File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        index = 0
        last_index = len(spaces)
        string = ''
        for i,c in enumerate(s):
            if(index < last_index and i == spaces[index]):
                string += ' '
                index+=1
            string += c
        
        return string